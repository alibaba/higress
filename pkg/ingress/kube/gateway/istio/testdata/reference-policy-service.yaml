apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: higress-system
spec:
  addresses:
  - value: higress-gateway
    type: Hostname
  gatewayClassName: higress
  listeners:
  - name: simple
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: ReferenceGrant
metadata:
  name: allow-service
  namespace: service
spec:
  from:
  - group: gateway.networking.k8s.io
    kind: HTTPRoute
    namespace: higress-system
  to:
  - group: ""
    kind: Service
    name: my-svc
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: higress-system
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  hostnames: ["simple.domain.example"]
  rules:
  - backendRefs:
    - name: my-svc
      namespace: service
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: backend-not-allowed
  namespace: higress-system
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  hostnames: ["simple2.domain.example"]
  rules:
  - backendRefs:
    - name: my-svc
      namespace: service
      port: 80
      weight: 1
    - name: httpbin
      namespace: default
      port: 80
      weight: 1
