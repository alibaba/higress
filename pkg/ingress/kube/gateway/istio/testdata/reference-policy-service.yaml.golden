apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/simple.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-simple
  namespace: higress-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/backend-not-allowed.higress-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: backend-not-allowed-0-istio-autogenerated-k8s-gateway
  namespace: higress-system
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-simple
  hosts:
  - simple2.domain.example
  http:
  - name: backend-not-allowed
    route:
    - destination:
        host: my-svc.service.svc.domain.suffix
        port:
          number: 80
      weight: 1
    - destination: {}
      weight: 1
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.higress-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: higress-system
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-simple
  hosts:
  - simple.domain.example
  http:
  - name: http
    route:
    - destination:
        host: my-svc.service.svc.domain.suffix
        port:
          number: 80
---
